### PR TITLE
[release/7.0-rc1] HTTP/3 flaky test retries and logging

### PR DESCRIPTION
**Pull request by JamesNK**
Address various pipeline failures:

- [Pipelines - Run 20220818.34 (azure.com)](https://dev.azure.com/dnceng/public/_build/results?buildId=1952772&view=ms.vss-test-web.build-test-results-tab)
- [Pipelines - Run 20220818.38 (azure.com)](https://dev.azure.com/dnceng/public/_build/results?buildId=1952956&view=ms.vss-test-web.build-test-results-tab)
- [Pipelines - Run 20220817.3 (azure.com)](https://dev.azure.com/dnceng/public/_build/results?buildId=1950541&view=ms.vss-test-web.build-test-results-tab)
- [Pipelines - Run 20220818.2 (azure.com)](https://dev.azure.com/dnceng/public/_build/results?buildId=1951376&view=ms.vss-test-web.build-test-results-tab)
- [Pipelines - Run 20220818.3 (azure.com)](https://dev.azure.com/dnceng/public/_build/results?buildId=1952786&view=ms.vss-test-web.build-test-results-tab&runId=50232550&resultId=109657&paneView=debug)

More retries will hopefully remove one set of failures. More logging to find solutions for others in the future. System.Net.Quic team is still making changes, so it can be a source of instability.

Test only changes.

### CODE REVIEW

File: src/Servers/Kestrel/Transport.Quic/test/QuicConnectionContextTests.cs
Comment (dougbu) [inline src/Servers/Kestrel/Transport.Quic/test/QuicConnectionContextTests.cs:424]: unreferenced❔

File: src/Servers/Kestrel/shared/test/ServerRetryHelper.cs
Comment (dougbu) [inline src/Servers/Kestrel/shared/test/ServerRetryHelper.cs:10]: nit: There's no delay in the loop below beyond connection timeouts. Suggest going higher if I'm right that address in use normally causes an almost-immediate failure.

File: src/Servers/Kestrel/test/Interop.FunctionalTests/Http3/Http3RequestTests.cs
Comment (dougbu) [inline src/Servers/Kestrel/test/Interop.FunctionalTests/Http3/Http3RequestTests.cs:231]: Unreferenced❔

**Review by dougbu — APPROVED** [review]
I'm liking this better than my #43422. Lemme know if there's anything I should keep there beyond the `StreamPool_StreamAbortedOnClientAndServer_NotPooled` retry. See also #43410 in case that also overlaps these fixes.

File: src/Servers/Kestrel/Transport.Quic/test/QuicConnectionContextTests.cs
Comment (JamesNK) [reply src/Servers/Kestrel/Transport.Quic/test/QuicConnectionContextTests.cs:424]: Correct. It registers an event source in the constructor and removes it on dispose. Since no one uses it, it could be changed to:

```cs
using var _ = new HttpEventSourceListener(LoggerFactory);
```

But I'd prefer to do that in a separate PR to remove them all at once. Personally, I'm happy leaving it as is.

**Comment by dougbu** [comment]
Any reason this hasn't been checked in @adityamandaleeka @JamesNK❔

**Comment by JamesNK** [comment]
Waiting on @adityamandaleeka approval 🙏

**Comment by adityamandaleeka** [comment]
Sorry, I hadn't seen this one. Approved, test-only changes.

**Comment by JamesNK** [comment]
I forgot to add you 😄

**Pull request merged**
